File: src/main/java/com/github/dylmeadows/eontimer/model/TimerSettingsModel.java
package com.github.dylmeadows.eontimer.model;

import javafx.beans.property.*;

@SuppressWarnings({"unused", "WeakerAccess"})
public class TimerSettingsModel {

    private final ObjectProperty<Console> console = new SimpleObjectProperty<>(TimerSettingsConstants.DEFAULT_CONSOLE);
    private final IntegerProperty refreshInterval = new SimpleIntegerProperty(TimerSettingsConstants.DEFAULT_REFRESH_INTERVAL);
    private final BooleanProperty precisionCalibrationMode = new SimpleBooleanProperty(TimerSettingsConstants.DEFAULT_PRECISION_CALIBRATION_MODE);
    private final transient IntegerProperty minimumLength = new SimpleIntegerProperty(TimerSettingsConstants.DEFAULT_MINIMUM_LENGTH);

    public Console getConsole() {
        return console.get();
    }

    public ObjectProperty<Console> consoleProperty() {
        return console;
    }

    public void setConsole(Console console) {
        this.console.set(console);
    }

    public int getRefreshInterval() {
        return refreshInterval.get();
    }

    public IntegerProperty refreshIntervalProperty() {
        return refreshInterval;
    }

    public void setRefreshInterval(int refreshInterval) {
        this.refreshInterval.set(refreshInterval);
    }

    public boolean isPrecisionCalibrationMode() {
        return precisionCalibrationMode.get();
    }

    public BooleanProperty precisionCalibrationModeProperty() {
        return precisionCalibrationMode;
    }

    public void setPrecisionCalibrationMode(boolean precisionCalibrationMode) {
        this.precisionCalibrationMode.set(precisionCalibrationMode);
    }

    public int getMinimumLength() {
        return minimumLength.get();
    }

    public IntegerProperty minimumLengthProperty() {
        return minimumLength;
    }

    public void setMinimumLength(int minimumLength) {
        this.minimumLength.set(minimumLength);
    }
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/settings/TimerSettingsModel.kt
package io.github.dylmeadows.eontimer.model.settings

import io.github.dylmeadows.eontimer.util.getValue
import io.github.dylmeadows.eontimer.util.setValue
import javafx.beans.property.BooleanProperty
import javafx.beans.property.LongProperty
import javafx.beans.property.ObjectProperty
import javafx.beans.property.SimpleBooleanProperty
import javafx.beans.property.SimpleIntegerProperty
import javafx.beans.property.SimpleLongProperty
import javafx.beans.property.SimpleObjectProperty

class TimerSettingsModel {
    val consoleProperty: ObjectProperty<Console> = SimpleObjectProperty(TimerSettingsConstants.DEFAULT_CONSOLE)
    var console: Console by consoleProperty

    val refreshIntervalProperty: LongProperty = SimpleLongProperty(TimerSettingsConstants.DEFAULT_REFRESH_INTERVAL)
    var refreshInterval by refreshIntervalProperty

    val precisionCalibrationModeProperty: BooleanProperty = SimpleBooleanProperty(TimerSettingsConstants.DEFAULT_PRECISION_CALIBRATION_MODE)
    var precisionCalibrationMode by precisionCalibrationModeProperty
}